src/dune_rules/preprocessing.ml

```diff
@@ -399,57 +400,60 @@ let get_cookies ~loc ~expander ~lib_name libs =
       , Some ("library-name", (library_name, Lib_name.of_local (loc, lib_name)))
       )
   in
-  match
-    List.concat_map libs ~f:(fun t ->
+  let+ cookies =
+    Memo.Build.List.concat_map libs ~f:(fun t ->
         let info = Lib.info t in
         let kind = Lib_info.kind info in
         match kind with
-        | Normal -> []
+        | Normal -> Memo.Build.return []
         | Ppx_rewriter { cookies }
         | Ppx_deriver { cookies } ->
-          List.map
+          Memo.Build.List.map
             ~f:(fun { Lib_kind.Ppx_args.Cookie.name; value } ->
-              (name, (Expander.Static.expand_str expander value, Lib.name t)))
+              let+ value = Expander.No_deps.expand_str expander value in
+              (name, (value, Lib.name t)))
             cookies)
-    |> (fun l ->
-         match library_name_cookie with
-         | None -> l
-         | Some cookie -> cookie :: l)
-    |> String.Map.of_list_reducei
-         ~f:(fun name ((val1, lib1) as res) (val2, lib2) ->
-           if String.equal val1 val2 then
-             res
-           else
-             let lib1 = Lib_name.to_string lib1 in
-             let lib2 = Lib_name.to_string lib2 in
-             User_error.raise ~loc
-               [ Pp.textf
-                   "%s and %s have inconsistent requests for cookie %S; %s \
-                    requests %S and %s requests %S"
-                   lib1 lib2 name lib1 val1 lib2 val2
-               ])
-    |> String.Map.foldi ~init:[] ~f:(fun name (value, _) acc ->
-           (name, value) :: acc)
-    |> List.rev
-    |> List.concat_map ~f:(fun (name, value) ->
-           [ "--cookie"; sprintf "%s=%S" name value ])
-  with
-  | x -> Resolve.return x
-  | exception User_error.E (msg, []) -> Resolve.fail msg
+  in
+  cookies
+  |> (fun l ->
+       match library_name_cookie with
+       | None -> l
+       | Some cookie -> cookie :: l)
+  |> String.Map.of_list_reducei
+       ~f:(fun name ((val1, lib1) as res) (val2, lib2) ->
+         if String.equal val1 val2 then
+           res
+         else
+           let lib1 = Lib_name.to_string lib1 in
+           let lib2 = Lib_name.to_string lib2 in
+           User_error.raise ~loc
+             [ Pp.textf
+                 "%s and %s have inconsistent requests for cookie %S; %s \
+                  requests %S and %s requests %S"
+                 lib1 lib2 name lib1 val1 lib2 val2
+             ])
+  |> String.Map.foldi ~init:[] ~f:(fun name (value, _) acc ->
```

Comment:
`String.Map.to_list_map` to simplify and get rid of the `List.rev`

Reply:
Well spotted, I updated the code